libflux/src/flux/semantic/builtins.rs

```diff
@@ -275,6 +275,267 @@ pub fn builtins() -> Builtins<'static> {
+                        ?desc: bool
+                    ) -> [t0]
+                "#),
+                "spread" => Node::Builtin(r#"
+                    forall [t0, t1] where t0: Row, t1: Row (
+                        <-tables: [t0],
+                        ?column: string
+                    ) -> [t1]
+                "#),
+                "stateTracking" => Node::Builtin(r#"
+                    forall [t0, t1] where t0: Row, t1: Row (
+                        <-tables: [t0],
+                        fn: (r: t0) -> bool,
+                        ?countColumn: string,
+                        ?durationColumn: string,
+                        ?durationUnit: duration,
+                        ?timeColumn: string
+                    ) -> [t1]
+                "#),
+                "stddev" => Node::Builtin(r#"
+                    forall [t0, t1] where t0: Row, t1: Row (
+                        <-tables: [t0],
+                        ?column: string,
+                        mode: string
+                    ) -> [t1]
+                "#),
+                "string" => Node::Builtin("forall [t0] (v: t0) -> string"),
+                "sum" => Node::Builtin(r#"
+                    forall [t0, t1] where t0: Row, t1: Row (
+                        <-tables: [t0],
+                        ?column: string
+                    ) -> [t1]
+                "#),
+                "tableFind" => Node::Builtin(r#"
+                    forall [t0, t1] where t0: Row, t1: Row (
+                        <-tables: [t0],
+                        fn: (key: t1) -> bool
+                    ) -> t0
+                "#),
+                "tail" => Node::Builtin(r#"
+                    forall [t0] (
+                        <-tables: [t0],
+                        n: int,
+                        ?offset: int
+                    ) -> [t0]
+                "#),
+                "time" => Node::Builtin("forall [t0] (v: t0) -> time"),
+                "timeShift" => Node::Builtin(r#"
+                    forall [t0] (
+                        <-tables: [t0],
+                        "duration": duration,
+                        ?columns: [string]
+                    ) -> [t0]
+                "#),
+                "tripleExponentialDerivative" => Node::Builtin(r#"
+                    forall [t0] where t0: Numeric, t1: Row (
+                        <-tables: [{_value: t0 | t1}],
+                        n: int
+                    ) -> [{_value: float | t1}]
+                "#),
+                "true" => Node::Builtin("forall [] bool"),
+                "uint" => Node::Builtin("forall [t0] (v: t0) -> uint"),
+                // This is suspicous: each input table stream is not restricted to the same schema.
+                // What's the right signature?
+                "union" => Node::Builtin(r#"
+                    forall [t0] where t0: Row, t1: Row (
+                        tables: [[t0]]
+                    ) -> [t1]
+                "#),
```

Comment:
I think they should be the same schema. From the type checker's point of view, each record type looks like this:

    { ... known labels ... | t0 } // where t0 is a row variable representing unknown labels

So if there is a contradiction on the known label types, type inference will error. But as long as there aren't any inconsistencies for the known label types, type inference won't care.

Reply:
Ok, so the type is fine as-is then, that's great.  I'll update the comment.

Reply:
I think we can even return `[t0]` as each table will be constrained to have the same schema.

Reply:
I'm a little confused by this, how about this example:
```
// type is [{host: string | t0}]:
ts0 = from(bucket: "foo") |> range(start: -5m) |> filter(fn: (r) => r.host == "bar")
// type is [{region: string | t0}]:
ts1 = from(bucket: "foo") |> range(start: -5m) |> filter(fn: (r) => r.region == "europe")

union(tables: [ts0, ts1])
```

Does the element type of the `tables` argument to `union` then become `{host: string | region: string | t0}`?  And does that mean that we assume that both `ts0` and `ts1` have a `region` and `host` field?  Because it's not necessarily the case.

Reply:
(all of these `t0`s are logically different in this example)

Reply:
That's right. So we will infer that the records from each table passed into `union` have the type:

    {host: string | region: string | t0}

You are also correct that the tables from `ts0` might not have a `region` column. This is fine because strings, like all column types, are nullable and so if the `region` column is referenced after the union, any null values will be passed through without error.

Reply:
Ok, that's great, I forgot about the `Nullable` kind.